### DISCUSSION TITLE
Step-like graphs for swimming fit files

### DISCUSSION THREAD

**Discussion by travelerdg** [mytourbook/mytourbook#1436] [General]
I don't succeed in getting step-like graphs (speed, cadence, stroke and swolf) for swimming fit files in MT. Tcx files work fine but only for speed and cadence data Changing the smooth parameter in the tour graph does nothing.
 
![Capture d’écran 2024-12-11 103902](https://github.com/user-attachments/assets/dd62f2e2-a6a1-4192-9baf-0d66c0f5af1d)

Maybe,, merging both files could give good results.

Thank you for your reply.

**Comment by wolfgang-ch** [comment]
Can you attach or send me the .fit file with swimming data, so I can test it

**Comment by travelerdg** [comment]
Herein  the swimming fit file.

[swimming_file.zip](https://github.com/user-attachments/files/18099074/swimming_file.zip)

**Comment by wolfgang-ch** [comment]
> I don't succeed in getting step-like graphs (speed, cadence, stroke and swolf) for swimming fit files in MT. 

The fit file you provided can display speed, stroke and swolf but not cadence

![image](https://github.com/user-attachments/assets/7803e401-3335-45d2-a33a-8eb6e3398c65)


> Tcx files work fine but only for speed and cadence data Changing the smooth parameter in the tour graph does nothing.

I don't understand this part of your sentence

**Comment by travelerdg** [comment]
Surprising, I've just checked the data included in a csv file from Fit SDK. Speed, strokes and cadence are well recorded then the Swolf is calculated by any softwares. For the same file, I obtained the expected behavior  from a more specific softawre, namely TrainingLab Pro :

![Screenshot](https://github.com/user-attachments/assets/c55af0fc-0936-4b0d-a048-7068322174a2)

It is possible to convert a fit file in tcx file with TrainingLab Pro (As MT does it). I've tested thie tcx file in MT. It gives the expected graphs but only for speed and cadence (strokes is not included in the tcx format).
  
[2024-11-15-10-28-44_FFRT_SDK_data.csv](https://github.com/user-attachments/files/18128435/2024-11-15-10-28-44_FFRT_SDK_data.csv)

**Reply by wolfgang-ch** [reply]
When the stroke values are not in the tcx file, then you need in MT an import of csv files, e.g. the above file, which is containing stroke values. Have I understood this correctly?

But why do you need a CSV importer when you already have a .fit file?

**Comment by travelerdg** [comment]
I only used the CSV importer to check the cadence values (Ive the latest version of MT and cadence is well displayed above in my first screenshot). More generally, it seems to me that the problem is related to the smoothing of graphs without gps data. 

The swimming files are recorded per unit of lenght not per unit of time. Can you tell me if, MT uses a smoothing algorithm for the graphs which only depends on time. ?

**Comment by wolfgang-ch** [comment]
> ... it seems to me that the problem ...

I've not yet understood what the problem is


I just discovered, that the cadence values will only be displayed when the tour is saved and not just imported, but I think this is not your question.

![image](https://github.com/user-attachments/assets/ab0aa9a3-5b09-40ac-9e2a-7a9d27104863)

**Comment by travelerdg** [comment]
Amazing software !

My question is :, why does not the smooth parameters in the tour graph modify the shape of the curves for swimming data ?

I hope now it is more understandable.

Thank you for your reply

**Comment by wolfgang-ch** [comment]
>...why does not the smooth parameters in the tour graph modify the shape of the curves for swimming data ?

Now I understand your question and now also the title of this discussion but these are two different things:

* Smooth swim values, e.g. elevation
* Display swim values with bars like in your graph https://github.com/mytourbook/mytourbook/discussions/1436#discussioncomment-11558819

Can you create 2 feature request for them

**Comment by travelerdg** [comment]
I've just created the two feature requests with screenshots

Regards

**Reply by wolfgang-ch** [reply]
These are the links https://github.com/mytourbook/mytourbook/issues/1438 https://github.com/mytourbook/mytourbook/issues/1439
